app/viewmodels/v2/SummaryCardHelper.scala

```diff
@@ -0,0 +1,420 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels.v2
+
+import models.User
+import models.v3.appealInfo.AppealStatusEnum
+import models.v3.lpp.{LPPDetails, LPPPenaltyCategoryEnum, LPPPenaltyStatusEnum}
+import models.v3.lsp.TaxReturnStatusEnum.Fulfilled
+import models.v3.lsp.{LSPDetails, LSPPenaltyCategoryEnum, LSPPenaltyStatusEnum, LateSubmission, TaxReturnStatusEnum}
+import play.api.i18n.Messages
+import play.twirl.api.{Html, HtmlFormat}
+import uk.gov.hmrc.govukfrontend.views.viewmodels.content.{HtmlContent, Text}
+import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist.{Key, SummaryListRow, Value}
+import uk.gov.hmrc.govukfrontend.views.viewmodels.tag.Tag
+import utils.{CurrencyFormatter, ImplicitDateFormatter, ViewUtils}
+import utils.v2.PenaltyPeriodHelper
+
+import java.time.LocalDate
+import javax.inject.Inject
+
+class SummaryCardHelper @Inject()(link: views.html.components.link) extends ImplicitDateFormatter with ViewUtils{
+
+  def populateLateSubmissionPenaltyCard(penalties: Seq[LSPDetails],
+                                        threshold: Int, activePoints: Int)
+                                       (implicit messages: Messages, user: User[_]): Seq[LateSubmissionPenaltySummaryCard] = {
+    val thresholdMet: Boolean = pointsThresholdMet(threshold, activePoints)
+    val filteredActivePenalties: Seq[LSPDetails] = penalties.filter(_.penaltyStatus != LSPPenaltyStatusEnum.Inactive).reverse
+    val indexedActivePoints = filteredActivePenalties.zipWithIndex
+    penalties.map { penalty =>
+      val newPenalty = findAndReindexPointIfIsActive(indexedActivePoints, penalty)
+      (newPenalty.penaltyCategory, newPenalty.penaltyStatus, newPenalty.appealInformation) match {
+        case (LSPPenaltyCategoryEnum.Charge | LSPPenaltyCategoryEnum.Threshold, _, _) => financialSummaryCard(newPenalty, threshold)
+        //case (LSPPenaltyCategoryEnum.Point, LSPPenaltyCategoryEnum.Added, _) => addedPointCard(newPenalty, thresholdMet) TODO: get mapping for LSPPenaltyCategoryEnum.Added
+        case (LSPPenaltyCategoryEnum.Point, LSPPenaltyStatusEnum.Inactive, None) => removedPointCard(newPenalty)
+        case (LSPPenaltyCategoryEnum.Point, _, _) => pointSummaryCard(newPenalty, thresholdMet)
+        case _ => throw new Exception("[SummaryCardHelper][populateLateSubmissionPenaltyCard] Incorrect values provided")
+      }
+    }
+  }
```

Comment:
Code to determine if point has been added, requires FAPIndicator to be made optional, not sure this would allow removed points to be handled correctly so may require further refactoring
```suggestion
  def populateLateSubmissionPenaltyCard(penalties: Seq[LSPDetails],
                                        threshold: Int, activePoints: Int)
                                       (implicit messages: Messages, user: User[_]): Seq[LateSubmissionPenaltySummaryCard] = {
    val thresholdMet: Boolean = pointsThresholdMet(threshold, activePoints)
    val filteredActivePenalties: Seq[LSPDetails] = penalties.filter(_.penaltyStatus != LSPPenaltyStatusEnum.Inactive).reverse
    val indexedActivePoints = filteredActivePenalties.zipWithIndex
    penalties.map { penalty =>
      val newPenalty = findAndReindexPointIfIsActive(indexedActivePoints, penalty)
      (newPenalty.penaltyCategory, newPenalty.penaltyStatus, newPenalty.appealInformation, newPenalty.FAPIndicator) match {
        case (LSPPenaltyCategoryEnum.Point, _, _, Some("X")) => addedPointCard(newPenalty, thresholdMet)
        case (LSPPenaltyCategoryEnum.Charge | LSPPenaltyCategoryEnum.Threshold, _, _, _) => financialSummaryCard(newPenalty, threshold)
        case (LSPPenaltyCategoryEnum.Point, LSPPenaltyStatusEnum.Inactive, None, _) => removedPointCard(newPenalty)
        case (LSPPenaltyCategoryEnum.Point, _, _, _) => pointSummaryCard(newPenalty, thresholdMet)
        case _ => throw new Exception("[SummaryCardHelper][populateLateSubmissionPenaltyCard] Incorrect values provided")
      }
    }
  }
```